### ISSUE TITLE
Delay JS - Ocean Elementor Widgets - WooCommerce product gallery zoom isn't working

### ISSUE THREAD

**Issue by vmanthos** [wp-media/wp-rocket#4271]
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version ✅ `3.9.1.1`
- Used the search feature to ensure that the bug hasn’t been reported before ✅

**Describe the bug**

When the [Ocean Elementor Widgets](https://oceanwp.org/extension/ocean-elementor-widgets/) plugin is used, despite the default WooCommerce`, exclusions:
https://github.com/wp-media/wp-rocket/blob/0413fa410ea220213ca98dc30c7edaf738cbf67f/inc/ThirdParty/Plugins/Ecommerce/WooCommerceSubscriber.php#L565-L569

the zoom functionality in the product gallery isn't working. 
More specifically, the zoom icon isn't displayed:
![](https://i.vgy.me/dg2aU2.jpg)

Excluding the following file fixed the issue:
```
/ocean-elementor-widgets/assets/js/vendors/photoswipe.min.js
```

**To Reproduce**
1. Install the [Ocean Elementor Widgets](https://oceanwp.org/extension/ocean-elementor-widgets/) plugin - this is a premium one.
2. Check a product that has a product gallery. 
3. The zoom icon will be missing.

**Expected behavior**

Zoom should be working.

**Additional context**

In the default exclusions, with the current `jQuery` RegEx, we are not picking `jquery.js`.
If we don't change the RegEx, `jquery.js` should be added in the `$exclusions`.

**Ticket:** https://secure.helpscout.net/conversation/1606049733/288598/

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

**Comment by Mai-Saad** [comment]
Note:
- Couldn't reproduce it visually without the premium plugin.
- For retest fix: we need to confirm that the mentioned script is excluded from Delay JS

**Comment by mostafa-hisham** [comment]
### Reproduce the issue :heavy_multiplication_x: 
I don't have Ocean Elementor Widgets plugin

### Identify the root cause ✅ 
The script should not be delayed to avoid this problem
But I don't know if should add it in WooCommerceSubscriber.php
i think it is better to exclude it in elementor`inc/ThirdParty/Plugins/PageBuilder/Elementor.php`
or create a new ThirdParty compatibility class

### Scope a solution ✅ 
for new ThirdParty compatibility class
We need to create a new plugin ThirdParty compatibility class for  Ocean Elementor Widgets
in "get_subscribed_events" check if the plugin is active and add
`$events['rocket_delay_js_exclusions'] = 'exclude_js_from_delay_js';`

```
public function show_notempty_product_gallery_with_delayJS( array $exclusions = [] ) {
	$exclusions[] = '/ocean-elementor-widgets/assets/js/vendors/photoswipe.min.js';
	return $exclusions;
}
```
Create Unit and integration tests for the new class


### Estimate the effort ✅
Effort `[S]`

**Comment by CrochetFeve0251** [comment]
@vmanthos I will fix the issue with unit and integration testing checking that it excludes the file beacuse I don't have the plugin.
However if you have the plugin share it to me so I can check if the solution doesn't produce another JS error

**Comment by vmanthos** [comment]
@CrochetFeve0251 Unfortunately, I don't have that plugin.

**Comment by CrochetFeve0251** [comment]
@vmanthos Nvm I tested the exclusion and I think that gonna be enough

**Comment by piotrbak** [comment]
Will be fixed in the 3.13 backend

**Comment by piotrbak** [comment]
Handled in 3.13 using dynamic lists
